src/mimir_coordinator.py

```diff
@@ -81,49 +86,130 @@ def is_recommended(self) -> bool:
+    # The ruler looks for tenant rules in the self._root_data_dir/rules/<TENANT ID> directory. The ruler requires rule files to be in the Prometheus format.
+    def _build_ruler_storage_config(self) -> Dict[str, Any]:
+        return {
+            "filesystem": {
+                "dir": str(self._root_data_dir / "rules"),
             },
         }
 
-        if self._s3_requirer:
-            s3_config = self._s3_requirer.s3_config
-            mimir_config["common"]["storage"] = {
-                "backend": "s3",
-                "s3": {
-                    "region": s3_config.region,  # eg. 'us-west'
-                    "bucket_name": s3_config.bucket_name,  # eg: 'mimir'
-                },
-            }
-            mimir_config["blocks_storage"] = {
-                "s3": {"bucket_name": s3_config.blocks_bucket_name}  # e.g. 'mimir-blocks'
-            }
-
-        # memberlist config for gossip and hash ring
-        mimir_config["memberlist"] = {
-            "join_members": list(self._cluster_provider.gather_addresses())
+    # bucket_store: sync_dir
+    # Directory to store synchronized TSDB index headers. This directory is not
+    # required to be persisted between restarts, but it's highly recommended
+
+    # filesystem: dir
+    # Mimir upload blocks (of metrics) to the object storage at period interval.
+
+    # tsdb: dir
+    # Directory to store TSDBs (including WAL) in the ingesters.
+    #  This directory is required to be persisted between restarts.
+
+    # The TSDB dir is used by ingesters, while the filesystem: dir is the "object storage"
+    # Ingesters are expected to upload TSDB blocks to filesystem: dir every 2h.
+    def _build_blocks_storage_config(self) -> Dict[str, Any]:
+        return {
+            "bucket_store": {
+                "sync_dir": str(self._root_data_dir / "tsdb-sync"),
+            },
+            "filesystem": {
+                "dir": str(self._root_data_dir / "blocks"),
+            },
+            "tsdb": {
+                "dir": str(self._root_data_dir / "tsdb"),
+            },
         }
 
-        # todo: TLS config for memberlist
-        if self._tls_requirer:
-            mimir_config["tls_enabled"] = True
-            mimir_config["tls_cert_path"] = self._tls_requirer.cacert
-            mimir_config["tls_key_path"] = self._tls_requirer.key
-            mimir_config["tls_ca_path"] = self._tls_requirer.capath
+    def _build_s3_storage_config(self, s3_data: _S3StorageBackend) -> Dict[str, Any]:
+        return {
+            "backend": "s3",
+            "s3": {
+                "endpoint": s3_data.endpoint,
+                "access_key_id": s3_data.access_key,
+                "secret_access_key": s3_data.secret_key,
+                "bucket_name": s3_data.bucket,
+                "region": s3_data.region,
+            },
+        }
 
-        return mimir_config
+    def _update_s3_storage_config(
```

Comment:
not sure what this method is doing and why

Reply:
added some docstrings to explain, this mainly pops any occurrence of filesystem in the mimir config and replaces it with a storage_prefix to be used with the s3 bucket and it only gets triggered if the S3 pydantic model has actual values and not defaults